src/components/document/annotation-layer.tsx

```diff
@@ -0,0 +1,173 @@
+    const height = objectBoundingBox.height;
+    const width = objectBoundingBox.width;
+    return { left, top, height, width };
+  }
+  
+  function getObjectBoundingBoxUnknownRow(
+    tileId: string, objectId: string
+  ) {
+    if (!content) return undefined;
+  
+    const rowId = content.findRowContainingTile(tileId);
+    return getObjectBoundingBox(rowId ?? "", tileId, objectId);
+  }
+
+  const handleAnnotationButtonClick = (tileId: string, objectId: string, objectType?: string) => {
+    if (!sourceTileId || !sourceObjectId) {
+      // We don't have a source object yet, so make this one the source object
+      setSourceTileId(tileId);
+      setSourceObjectId(objectId);
+      setSourceObjectType(objectType);
+    } else if (tileId === sourceTileId && objectId === sourceObjectId && objectType === sourceObjectType) {
+      // This object is already selected as the source object, so deselect it
+      setSourceTileId("");
+      setSourceObjectId("");
+      setSourceObjectType(undefined);
+    } else {
+      // Create an arrow from the source object to this object
+      const sourceObject =
+        ClueObjectModel.create({ tileId: sourceTileId, objectId: sourceObjectId, objectType: sourceObjectType });
+      const targetObject = ClueObjectModel.create({ tileId, objectId, objectType });
+      content?.addArrow(ArrowAnnotation.create({ sourceObject, targetObject }));
+      setSourceTileId("");
+      setSourceObjectId("");
+      setSourceObjectType(undefined);
+    }
+  };
+
+  const getBoundingBox = (object: ClueObjectType) => {
+    return getObjectBoundingBoxUnknownRow(object.tileId, object.objectId);
+  };
+
+  const editing = ui.annotationMode !== undefined;
+  const hidden = !ui.showAnnotations;
+  const classes = classNames("annotation-layer", { editing, hidden });
+  return (
+    <div className={classes}>
+      { editing && !readOnly && rowIds.map(rowId => {
+        const row = content?.rowMap.get(rowId);
+        if (row) {
+          const tiles = row.tiles;
+          return tiles.map(tileInfo => {
+            const tile = content?.tileMap.get(tileInfo.tileId);
+            if (tile) {
+              return tile.content.annotatableObjects.map(({ objectId, objectType }) => {
+                return (
+                  <AnnotationButton
+                    getObjectBoundingBox={getObjectBoundingBox}
+                    key={`${tile.id}-${objectId}-button`}
+                    objectId={objectId}
+                    objectType={objectType}
+                    onClick={handleAnnotationButtonClick}
+                    rowId={rowId}
+                    sourceObjectId={sourceObjectId}
+                    sourceTileId={sourceTileId}
+                    tileId={tile.id}
+                  />
+                );
+              });
+            }
+          });
+        }
+      })}
+      <svg xmlnsXlink="http://www.w3.org/1999/xlink" className="annotation-svg" height="1500" width="1500">
```

Comment:
Should the `height` and `width` be dynamic based on the size of the component or viewport? Also, I probably would have rendered each arrow as a separate SVG, without having any particular reason for doing so. Did you have a particular reason for choosing to render all of the arrows as a single SVG?

Reply:
I didn't have a reason to use one SVG, it just seemed natural to me and I didn't consider doing a bunch of separate ones.

This hard coded size was definitely on my radar, and I thought it might be a problem, especially for very tall documents. Is there an easy way to figure this information out? Seems like you could use all of the row heights to determine the correct height. I'm not sure how to determine the correct width, but maybe the correct width doesn't matter as much?

Reply:
Turns out this was as easy as setting `height` and `width` to `100%`!